### PR TITLE
WIP: Adding Shadow effect on UWP

### PR DESCRIPTION
**Pull request by jhonatanfernando**
For now it's working only when the CornerRadius property has the same value for all sides.

### CODE REVIEW

**Comment by RyanGaudion** [comment]
@sthewissen is this something that could be added?

**Comment by sthewissen** [comment]
Sure, it's just that the WIP gave me the impression the PR wasn't done yet :)

**Comment by sthewissen** [comment]
Also, it's pretty hard for me to test seeing as how I don't have a Windows machine to test this with 😅

**Comment by jhonatanfernando** [comment]
It is not working for all situations yet, only when the CornerRadious have the same value. 


Sent from Yahoo Mail for iPhone


On Tuesday, January 14, 2020, 5:42 AM, Steven Thewissen <notifications@github.com> wrote:


Sure, it's just that the WIP gave me the impression the PR wasn't done yet :)

**Comment by RyanGaudion** [comment]
Oh okay. Thanks for the update. @jhonatanfernando , how long do you think until the pr will be ready?

**Comment by jhonatanfernando** [comment]
> Oh okay. Thanks for the update. @jhonatanfernando , how long do you think until the pr will be ready?

@RyanGaudion  @sthewissen Actually I don't know if will be possible to make it works in all possibilities. I am searching about this.  @sthewissen Will you be able to test these changes on Windows Device ?

**Comment by sthewissen** [comment]
@jhonatanfernando Not on my own laptop atm as I'm lacking a Windows instance. I'll see what I can do.

**Comment by jhonatanfernando** [comment]
> @jhonatanfernando Not on my own laptop atm as I'm lacking a Windows instance. I'll see what I can do.

Hey @sthewissen. Do you have any update about the pull request?

**Comment by sthewissen** [comment]
Looking at this as we speak.

File: src/Xamarin.Forms.PancakeView.Multi/Platforms/UWP/PancakeViewRenderer.cs [hunk 1]
```diff
@@ -117,30 +132,55 @@ protected override void OnElementPropertyChanged(object sender, PropertyChangedE
...
         private void UpdateShadow(PancakeView pancake)
         {
-            if (Control != null)
+            //For now gets the shadow only when the CornerRadius has the same value for all sides. 
+            if (Control != null && pancake.HasShadow && pancake.Width > 0 && pancake.Height > 0 &&
+                pancake.CornerRadius.TopLeft == pancake.CornerRadius.BottomLeft &&
+                pancake.CornerRadius.BottomRight == pancake.CornerRadius.TopRight)
             {
-
+                rectangle.Fill = new SolidColorBrush(Windows.UI.Colors.Black);
```
Comment (sthewissen) [inline src/Xamarin.Forms.PancakeView.Multi/Platforms/UWP/PancakeViewRenderer.cs]: Would this work with transparent as a color so we're sure no black shines through? The sample app looks like this for me:

image

Comment (jhonatanfernando) [reply src/Xamarin.Forms.PancakeView.Multi/Platforms/UWP/PancakeViewRenderer.cs]: It is not working for combination with individual rounded corners yet. I'll check about to use transparent color.

Comment (sthewissen) [reply src/Xamarin.Forms.PancakeView.Multi/Platforms/UWP/PancakeViewRenderer.cs]: I think it's better to not apply the shadow at all unless all 4 radii are the same. I will update the documentation accordingly to clearly state that limitation. Should be a minor adjustment to the current `if` check that's already in place.

Comment (sthewissen) [reply src/Xamarin.Forms.PancakeView.Multi/Platforms/UWP/PancakeViewRenderer.cs]: Other than that it's fine! Thanks again for making this :)

**Comment by sthewissen** [comment]
So with #82 also being made, I checked this out again and did the small update I mentioned. However, it seems that when toggling HasShadow on and back off the shadow is not removed. I thought the code I added would fix that, but it doesn't seem to. Do you happen to know how to reset the shadow when the binding turns it off? @jhonatanfernando

**Comment by jhonatanfernando** [comment]
> So with #82 also being made, I checked this out again and did the small update I mentioned. However, it seems that when toggling HasShadow on and back off the shadow is not removed. I thought the code I added would fix that, but it doesn't seem to. Do you happen to know how to reset the shadow when the binding turns it off? @jhonatanfernando

@sthewissen I will check this for you.

**Comment by jhonatanfernando** [comment]
Now it is removing the shadow for UWP when it is off.

**Comment by sthewissen** [comment]
Also removed the black background color from the visual, because when using irregular corners after enabling/disabling shadow it would leave behind a black artifact. I think this should be good now!

**Pull request merged**